### ISSUE TITLE
importing cluster

### ISSUE THREAD

**Issue by phaneesh11** [rancher/rancher#25111]
*Describe your issue here*
When importing cluster, I am getting below error

w8bhshvgrlnh6xkgnbb8z59bdb2ftjwkvwb"
Jan 25 21:40:50 centos2 journal: time="2020-01-25T16:10:50Z" level=info msg="Connecting to proxy" url="wss://15.213.50.241:8443/v3/connect"
Jan 25 21:40:50 centos2 journal: time="2020-01-25T16:10:50Z" level=error msg="Failed to connect to proxy. Response status: 200 - 200 OK. Response body: node.management.cattle.io \"c-qt4cq/m-1b0bdf35d558\" not found" error="websocket: bad handshake"
Jan 25 21:40:50 centos2 journal: time="2020-01-25T16:10:50Z" level=error msg="Remotedialer proxy error" error="websocket: bad handshake"
Jan 25 21:40:52 centos2 journal: 2020-01-25 16:10:52.943287 W | etcdserver: read-only range request "key:\"/registry/services/endpoints/default/kubernetes\" " with result "range_response_count:1 size:217" took too long (261.003885ms) to execute

---
| Useful | Info |
| :-- | :-- |
|Versions|Rancher `v2.3.0` UI: `v2.3.15` |
|Route|`global-admin.clusters.index`|

**Comment by superseb** [comment]
Please describe what kind of cluster you are importing (what installer or hosted service, what version, how many nodes) and the specifications of the machine running Rancher (I assume its single install). If you have more logging from the `rancher/rancher` container, please use gist or pastebin to supply that if possible.

**Comment by bmdepesa** [comment]
Possibly the same as https://github.com/rancher/rancher/issues/25372

**Comment by jguidoux** [comment]
Hi, 
I have the same issue: I am not behind a proxy.
I installed the version of rancher: v2.3.5. I also tried with the 'stable' version.
I have a kubernetes cluster on virtualbox with 3 nodes. (1 master and 2 workers)
I followed this video to set up rancher : https://www.youtube.com/watch?v=-xEGoiCXavw
My errors in the agent are : 
`time="2020-02-24T23:55:26Z" level=error msg="Remotedialer proxy error" error="websocket: bad handshake"
time="2020-02-24T23:55:36Z" level=info msg="Connecting to wss://192.168.1.14/v3/connect with token wq65m9rqlgg6mqc68nbbx86jbzbc4fkjlt44rdv4zqdrkq28m6p4zg"
time="2020-02-24T23:55:36Z" level=info msg="Connecting to proxy" url="wss://192.168.1.14/v3/connect"
time="2020-02-24T23:55:36Z" level=error msg="Failed to connect to proxy. Response status: 200 - 200 OK. Response body: node.management.cattle.io \"c-xhckj/m-8b61e793cd3b\" not found" error="websocket: bad handshake"
time="2020-02-24T23:55:36Z" level=error msg="Remotedialer proxy error" error="websocket: bad handshake"
`
The IP address is correct.

**Comment by jguidoux** [comment]
I have many a difference. My VMs containing Kubernetes nodes have been created using Vagrant and Ansible.

**Comment by Kampe** [comment]
I show the same issues when using the API to create and import a cluster. My pods rarely stand up correctly and report up, only stating 

`level=error msg="Failed to connect to proxy. Response status: 200 - 200 OK. Response body: cluster not found" error="websocket: bad handshake"`

**Comment by superseb** [comment]
The only way to diagnose what is going on here if we can replicate the behavior seen. Please describe what kind of cluster you are importing/how it is created/how we can create it in a similar way to reproduce and any other information that helps in pin pointing what the cause of the issue is.

**Comment by Kampe** [comment]
Nothing crazy, default custom cluster - 

This is the API call I use for creating the cluster to give you an idea of what kind of cluster we're trying to import (a custom k3os cluster)

```
CLUSTER_RESPONSE=$(curl -s "https://$RANCHER_URI/v3/cluster" -H 'content-type: application/json' -H "Authorization: Bearer $API_TOKEN" --data-binary '{"type": "cluster", "name": "'$CLUSTER_NAME'", "enableClusterAlerting":true, "enableClusterMonitoring":true}')
```

**Comment by Ericzzt** [comment]
I met the same issue. I have set up my own single node kubernetes cluster and tried to import it to the Rancher. the version of rancher is V 2.4.2. Kubernetes version 1.18.

**Comment by Ericzzt** [comment]
> Hi,
> I have the same issue: I am not behind a proxy.
> I installed the version of rancher: v2.3.5. I also tried with the 'stable' version.
> I have a kubernetes cluster on virtualbox with 3 nodes. (1 master and 2 workers)
> I followed this video to set up rancher : https://www.youtube.com/watch?v=-xEGoiCXavw
> My errors in the agent are :
> `time="2020-02-24T23:55:26Z" level=error msg="Remotedialer proxy error" error="websocket: bad handshake" time="2020-02-24T23:55:36Z" level=info msg="Connecting to wss://192.168.1.14/v3/connect with token wq65m9rqlgg6mqc68nbbx86jbzbc4fkjlt44rdv4zqdrkq28m6p4zg" time="2020-02-24T23:55:36Z" level=info msg="Connecting to proxy" url="wss://192.168.1.14/v3/connect" time="2020-02-24T23:55:36Z" level=error msg="Failed to connect to proxy. Response status: 200 - 200 OK. Response body: node.management.cattle.io \"c-xhckj/m-8b61e793cd3b\" not found" error="websocket: bad handshake" time="2020-02-24T23:55:36Z" level=error msg="Remotedialer proxy error" error="websocket: bad handshake" `
> The IP address is correct.

Hi, have you resolved this problem?

**Comment by Ericzzt** [comment]
I have met the same issue as above. I installed the Kubernetes cluster on single node by kubeadm. the version is 1.18. the rancher version is 2.4.2.  Could anyone help me with that problem

**Comment by jugou28** [comment]
same issue for me. No idea ?

**Comment by jugou28** [comment]
I resolve my problem :  
My Kubernetes daemon was in V1.15.12. But i did not apply "kubeadm upgrade apply"

Upgrade to the latest version in the v1.15 series:
COMPONENT            CURRENT   AVAILABLE
API Server           v1.15.9   v1.15.12
Controller Manager   v1.15.9   v1.15.12
Scheduler            v1.15.9   v1.15.12
Kube Proxy           v1.15.9   v1.15.12
CoreDNS              1.3.1     1.3.1
Etcd                 3.3.10    3.3.10


Now, my cluster is ready in rancher.

**Comment by peetasan** [comment]
Same issue here.

Rancher v2.4.3 
K8s v1.17.5
running on AWS on 3 t3.medium instances (as a test cluster) with Ubuntu 18.04.2

Trying to import a cluster from GCP, version v1.16.8-gke.15, 3 n1-standard-1 nodes running Ubuntu 18.04.4

Connectivity and DNS between the two is fine (through vpn tunneling)

logs from `cattle-node-agent`
```
time="2020-06-09T14:41:13Z" level=info msg="Connecting to wss://mydomain/v3/connect with token xxx"
time="2020-06-09T14:41:13Z" level=info msg="Connecting to proxy" url="wss://mydomain/v3/connect"
time="2020-06-09T14:41:13Z" level=error msg="Failed to connect to proxy. Response status: 200 - 200 OK. Response body: node.management.cattle.io \"c-wjkc9/m-29ab531060fd\" not found" error="websocket: bad handshake"
time="2020-06-09T14:41:13Z" level=error msg="Remotedialer proxy error" error="websocket: bad handshake"
```

Response from `daemonset/nginx-ingress-controller`
```
10.64.0.82 - [10.64.0.82] - - [09/Jun/2020:13:42:08 +0000] "GET /v3/connect HTTP/1.1" 200 60 "-" "Go-http-client/1.1" 3562 0.003 [cattle-system-rancher-80] [] 10.42.1.6:80 60 0.000 200 795846c28b193ce34b0967d511d46a7d```

What other logs to check? Where can I find more info about what is `node.management.cattle.io` and that particular resource?

**Comment by sunausti** [comment]
same issue here. I'm using Vagrant deploy k8s cluster , the host running rancher . once import cluster from vagrant , the cattle-node-agent-nscdb seems issue, 

time="2020-06-10T05:35:01Z" level=info msg="Connecting to proxy" url="wss://10.67.103.178/v3/connect"
time="2020-06-10T05:35:01Z" level=error msg="Failed to connect to proxy. Response status: 200 - 200 OK. Response body: node.management.cattle.io \"c-lq7bn/m-ae78d5301f0b\" not found" error="websocket: bad handshake"

**Comment by cheessee** [comment]
Same issue here.

I federate Rancher2 with the gke cluster using the "import cluster option", with the insecure curl command option.

When cattle-system namespace is deployed, I see this on cattle-node-agent-xxxx pod logs:

`time="2020-06-29T11:12:54Z" level=info msg="Connecting to wss://XX.XX.XX.XX:8443/v3/connect with token <TOKEN>"
time="2020-06-29T11:12:54Z" level=info msg="Connecting to proxy" url="wss://XX.XX.XX.XX:8443/v3/connect"
time="2020-06-29T11:12:54Z" level=error msg="Failed to connect to proxy. Response status: 200 - 200 OK. Response body: node.management.cattle.io \"c-fghq5/m-c0c7c5545476\" not found" error="websocket: bad handshake"
time="2020-06-29T11:12:54Z" level=error msg="Remotedialer proxy error" error="websocket: bad handshake"`

Trying to add **1.14.10-gke.36**  gke version and running rancher2 on a google gce vm with this compose:

`version: "2.1"
services:
  rancherinstance:
    image: rancher/rancher:v2.4.5
    container_name: rancherserver
    ports:
      - "8080:80"
      - "8443:443"
    volumes:
      - /datadrive/onesaitplatform/rancher/data:/var/lib/mysql
      - /datadrive/onesaitplatform/rancher/rancher:/var/lib/rancher
`
I also tried with version **v2.2.10** and **v2.4.2** according with Rancher2 version matrix, but got the same error.

https://rancher.com/support-maintenance-terms/all-supported-versions/rancher-v2.2.10/

I do the same process with AKS and I have **no problems**.

I can share more information if necessary, I've been trying to fix this for several days and can't find many solutions ...

Thanks!

**Comment by StardustXiaoT** [comment]
Rancher 2.4.3
K8s 1.18
I have the same issue. I built single node rke k8s. And I ran rancher in docker. 
First, I ran rancher in docker without proxy environment and certificaite(using rancher default ). It was fine to imporot rke k8s into rancher. 
However, once I set proxy and with cerfication.
Error:
msg="Failed to connect to proxy. Response status: 200 - 200 OK. Response body: node.management.cattle.io \"c-g5dhg/m-b18b19b0b901\" not found" error="websocket: bad handshake"

**Comment by ahsar** [comment]
Rancher 2.4.3
K8s 1.18
it cost about 3 hour to waiting.
BUT speical is my machine use vpn ipsec to connect.
Same issue

**Comment by dhthakkar-conga** [comment]
Any update on this. i m also facing similar issue.

**Comment by jamesgetx** [comment]
same issue for me, but I resolve my problem after adding 10.43.0.1 to NO_PROXY(rancher server run in docker).

**Comment by dhthakkar-conga** [comment]
@jamesgetx , Exactly where do we need to add NO_PROXY : 10.43.0.1? Can you please specify file name and more description?

**Comment by superseb** [comment]
@dhthakkar-conga Are you in a proxied environment? The docs are here https://rancher.com/docs/rancher/v2.x/en/installation/other-installation-methods/single-node-docker/proxy/ or for HA you can follow the directives described on https://rancher.com/docs/rancher/v2.x/en/installation/options/chart-options/ and use the same values.

Anyone running into this issue, please describe the network environment the imported cluster is in (either proxy/VPN/private link or internet) and provide the full log of the `cattle-node-agent` (but redact the token and Rancher URL)

**Comment by kspradheep** [comment]
I have the same issue on importing GKE v1.15.12 cluster size 56 node count to Rancher HA v2.3.6 (hosted in AWS).

**Network environment:**
private link (GCP network --> internal Megaport network ---> AWS private route)

Notes: 

- I was able to import GKE v1.14.x cluster to the same cluster and network environment is similar. I tried  [cleaning up cluster nodes](https://rancher.com/docs/rancher/v2.x/en/cluster-admin/cleaning-cluster-nodes/#removing-rancher-components-from-imported-clusters) re-importing the cluster but didn't help much.

- Rancher has this env var
         - name: NO_PROXY
          value: 127.0.0.0/8,10.0.0.0/8,172.16.0.0/12,192.168.0.0/16

**cattle-node-agent logs:**
time="2020-08-28T02:07:32Z" level=info msg="Connecting to wss://rancher.mydomain.com/v3/connect with token xxxxxxxxxxx"
time="2020-08-28T02:07:32Z" level=info msg="Connecting to proxy" url="wss://rancher-nonprod.iherb.net/v3/connect"
time="2020-08-28T02:07:33Z" level=error msg="Failed to connect to proxy. Response status: 200 - 200 OK. Response body: node.management.cattle.io \"c-mxvl4/m-41f04cc39a05\" not found" error="websocket: bad handshake" 

@superseb highly appreciate your help.

**Comment by dhthakkar-conga** [comment]
Its resolved. 

Previously, We were using nginx-ingress and skipeed deployment of default traefik on k3s installtion.
`curl -sfL https://get.k3s.io | INSTALL_K3S_EXEC="--write-kubeconfig-mode 644 --datastore-endpoint <endPoint> \
--no-deploy servicelb --no-deploy traefik"  sh -s -` 

and looks like its causing this issue. When we switched back to default traefik with below command on k3s and after that issue resolved.
`curl -sfL https://get.k3s.io | INSTALL_K3S_EXEC="--write-kubeconfig-mode 644 --datastore-endpoint <endPoint>" sh -s -`

**Comment by kspradheep** [comment]
@dhthakkar-conga any ideas how to troubleshoot this issue in RKE Kubernetes, Helm Rancher deployment environment? Thanks!

**Comment by dhthakkar-conga** [comment]
> @dhthakkar-conga any ideas how to troubleshoot this issue in RKE Kubernetes, Helm Rancher deployment environment? Thanks!

I've not tried RKE yet.

**Comment by kspradheep** [comment]
@superseb anything you could help?

**Comment by kspradheep** [comment]
It's resolved for me by doing upgrade from v2.3.9 to v2.4.6. However, rancher cli v2.4.6 is not working when trying to add AD groups to cluster (not seeing this issue in UI).

`rancher clusters add-member-role <ad group name> cluster-owner`

`FATA[0005] Bad response statusCode [422]. Status [422 Unprocessable Entity]. Body: [baseType=error, code=InvalidBodyContent, message=must target a user [userId]/[userPrincipalId] OR a group [groupId]/[groupPrincipalId]] from [https://rancher.mydomain.com/v3/clusterroletemplatebindings]`

**Comment by soumyalj** [comment]
Reproduced the CLI issue above in version v2.4.6
**Steps:**
1. Enable AD auth in the setup as admin
2. Create a cluster 
3. Using CLI add a group to to the cluster
```
./rancher cluster add-member-role testgroup1 cluster-member
FATA[0000] Bad response statusCode [422]. Status [422 Unprocessable Entity]. Body: [message=must target a user  
[userId]/[userPrincipalId] OR a group [groupId]/[groupPrincipalId], baseType=error, code=InvalidBodyContent] from [https://50.116.6.27/v3/clusterroletemplatebindings] 
```
Note: The same issue is observed in rancher CLI v2.4.5
Adding a user to the cluster as a member/cluster-owner is functional in CLI v2.4.6
